internal/exec/stages/disks/disks.go

```diff
@@ -361,18 +379,30 @@ func translateCreateOptionSliceToStringSlice(opts []types.CreateOption) []string
 	return newOpts
 }
 
-func (s stage) readFilesystemType(fs types.Mount) (string, error) {
+func (s stage) readFilesystemType(fs types.Mount) (string, string, string, error) {
```

Comment:
Does a lot more the reading the type. It's difficult to figure out which of these strings is which. Either return a struct, use named variables, or split this into multiple functions.